### PR TITLE
system test com_templates webservices

### PR DESCRIPTION
**Pull request by alikon**
### Summary of Changes
API tests for following endpoints:

- GET: /api/index.php/v1/templates/styles/site
- GET: /api/index.php/v1/templates/styles/site/{id}
- PATCH: /api/index.php/v1/templates/styles/site/{id}

### CODE REVIEW

File: api/components/com_templates/src/Controller/StylesController.php [hunk 1]
```diff
@@ -88,7 +88,7 @@ protected function preprocessSaveData(array $data): array
         // If we are updating an item the template is a readonly property based on the ID
         if ($this->input->getMethod() === 'PATCH') {
             if (\array_key_exists('template', $data)) {
-                throw new InvalidParameterException('The template property cannot be modified for an existing style');
```
Comment (laoneo) [inline api/components/com_templates/src/Controller/StylesController.php:91]: Why did you change this?

Comment (alikon) [reply api/components/com_templates/src/Controller/StylesController.php:91]: cause of  

> If we are updating an item the template is a readonly property based on the ID

Comment (laoneo) [reply api/components/com_templates/src/Controller/StylesController.php:91]: Can you revert that line and make a new pr against 5.0? The tests I would like to have in 4.4, but this change is a bc break.

Comment (alikon) [reply api/components/com_templates/src/Controller/StylesController.php:91]: it is not, it is a bug fix imho
look at https://github.com/joomla/joomla-cms/blob/4.4-dev/libraries/src/MVC/Controller/ApiController.php#L397-L406

in this case the 'template' key is always retrieved  even when it is not passed in body /payload 

so it have never worked with the PATCH

Comment (laoneo) [reply api/components/com_templates/src/Controller/StylesController.php:91]: Ok, got it as I could reproduce the issue on my dev server. Thanks for insisting!

Comment (alikon) [reply api/components/com_templates/src/Controller/StylesController.php:91]: i was trapped on this for solve #41681 😢 

thank you for listening

**Comment by HLeithner** [comment]
That silent error seems wrong to me, we shouldn't ignore modification requests without noticed, actually the exception is right from my point of view. the `patch` request shouldn't include a the `template` in the first place

**Comment by alikon** [comment]
the problem is that even if you don't include `template` in the payload of the PATCH request
as per https://github.com/joomla/joomla-cms/blob/4.4-dev/libraries/src/MVC/Controller/ApiController.php#L397-L406
it's included as well, there are some keys like `template` and or `password` that for PATCH needs some extra management
done in the `preprocessSaveData`
